reporter/internal/pdata/generate.go

```diff
@@ -0,0 +1,271 @@
+					mapping.SetFileOffset(traceInfo.MappingFileOffsets[i])
+					mapping.SetFilenameStrindex(getStringMapIndex(stringMap, fileName))
+
+					// Once SemConv and its Go package is released with the new
+					// semantic convention for build_id, replace these hard coded
+					// strings.
+					attrMgr.AppendOptionalString(mapping.AttributeIndices(),
+						"process.executable.build_id.gnu", ei.GnuBuildID)
+					attrMgr.AppendOptionalString(mapping.AttributeIndices(),
+						"process.executable.build_id.htlhash", traceInfo.Files[i].StringNoQuotes())
+				}
+				loc.SetMappingIndex(locationMappingIndex)
+			case libpf.AbortFrame:
+				// Next step: Figure out how the OTLP protocol
+				// could handle artificial frames, like AbortFrame,
+				// that are not originated from a native or interpreted
+				// program.
+			default:
+				// Store interpreted frame information as a Line message:
+				line := loc.Line().AppendEmpty()
+
+				fileIDInfoLock, exists := p.Frames.GetAndRefresh(traceInfo.Files[i],
+					FramesCacheLifetime)
+				if !exists {
+					// At this point, we do not have enough information for the frame.
+					// Therefore, we report a dummy entry and use the interpreter as filename.
+					line.SetFunctionIndex(createFunctionEntry(funcMap,
+						"UNREPORTED", frameKind.String()))
+				} else {
+					fileIDInfo := fileIDInfoLock.RLock()
+					if si, exists := (*fileIDInfo)[traceInfo.Linenos[i]]; exists {
+						line.SetLine(int64(si.LineNumber))
+
+						line.SetFunctionIndex(createFunctionEntry(funcMap,
+							si.FunctionName, si.FilePath))
+					} else {
+						// At this point, we do not have enough information for the frame.
+						// Therefore, we report a dummy entry and use the interpreter as filename.
+						// To differentiate this case from the case where no information about
+						// the file ID is available at all, we use a different name for reported
+						// function.
+						line.SetFunctionIndex(createFunctionEntry(funcMap,
+							"UNRESOLVED", frameKind.String()))
+					}
+					fileIDInfoLock.RUnlock(&fileIDInfo)
+				}
+
+				// To be compliant with the protocol, generate a dummy mapping entry.
+				loc.SetMappingIndex(getDummyMappingIndex(fileIDtoMapping, stringMap,
+					attrMgr, profile, traceInfo.Files[i]))
+			}
+		}
+
+		attrMgr.AppendOptionalString(sample.AttributeIndices(),
+			semconv.ContainerIDKey, traceKey.ContainerID)
+		attrMgr.AppendOptionalString(sample.AttributeIndices(),
+			semconv.ProcessCommandKey, traceKey.Comm)
+		attrMgr.AppendOptionalString(sample.AttributeIndices(),
+			semconv.ServiceNameKey, traceKey.ApmServiceName)
+		attrMgr.AppendInt(sample.AttributeIndices(),
+			semconv.ProcessPIDKey, traceKey.Pid)
+
+		if p.ExtraSampleAttrProd != nil {
+			extra := p.ExtraSampleAttrProd.ExtraSampleAttrs(attrMgr, traceKey.ExtraMeta)
+			sample.AttributeIndices().Append(extra...)
+		}
+
+		sample.SetLocationsLength(int32(len(traceInfo.FrameTypes)))
```

Comment:
This is incorrect, as frames of type `libpf.AbortFrame` are skipped. The correct value would be the number of interpreted frames + number of native frames.
But I think this is also a bug in the current implementation.

Reply:
Yes, the main branch has the same logic.
https://github.com/open-telemetry/opentelemetry-ebpf-profiler/blob/9eb454541a3cf1282a8755dbcec9a9471b243e83/reporter/otlp_reporter.go#L672

Let's open an external issue about this.

Reply:
See https://github.com/open-telemetry/opentelemetry-ebpf-profiler/issues/262